coderd/workspace_apps_internal_test.go

```diff
@@ -0,0 +1,100 @@
+package coderd
+
+import (
+	"context"
+	"crypto/sha256"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/coderd/database"
+	"github.com/coder/coder/coderd/database/databasefake"
+	"github.com/coder/coder/testutil"
+)
+
+func TestAPIKeyEncryption(t *testing.T) {
+	t.Parallel()
+
+	generateAPIKey := func(t *testing.T, db database.Store) (keyID, keySecret string, hashedSecret []byte, data encryptedAPIKeyPayload) {
+		keyID, keySecret, err := generateAPIKeyIDSecret()
+		require.NoError(t, err)
+
+		hashedSecretArray := sha256.Sum256([]byte(keySecret))
+		data = encryptedAPIKeyPayload{
+			APIKey:    keyID + "-" + keySecret,
+			ExpiresAt: database.Now().Add(24 * time.Hour),
+		}
+
+		return keyID, keySecret, hashedSecretArray[:], data
+	}
+	insertAPIKey := func(t *testing.T, db database.Store, keyID string, hashedSecret []byte) {
+		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
+		defer cancel()
+
+		_, err := db.InsertAPIKey(ctx, database.InsertAPIKeyParams{
+			ID:           keyID,
+			HashedSecret: hashedSecret,
+		})
+		require.NoError(t, err)
+	}
+
+	t.Run("OK", func(t *testing.T) {
+		t.Parallel()
+		db := databasefake.New()
+		keyID, _, hashedSecret, data := generateAPIKey(t, db)
+		insertAPIKey(t, db, keyID, hashedSecret)
+
+		encrypted, err := encryptAPIKey(data)
```

Comment:
What are your thoughts on moving `encryptAPIKey` and `decryptAPIKey` into a new package like `appwildcard` or something and moving associated helper functions there? I like all the code, but I'm worried that someone exploring the `coderd` package may not understand when or why we need to encrypt/decrypt keys.

Reply:
A problem we had in v1 was that the devurls code was split across many different places. I'd like to avoid that if possible in v2 by keeping all of the code in one place.

Would a comment above the functions resolve this?

Reply:
Fair enough, that code wasn't consolidated well. Seems good to me!